src/test/java/uk/gov/ons/census/action/schedule/FulfilmentsUpdatedTest.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.ons.census.action.schedule;
+
+import static org.mockito.Mockito.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.junit.Test;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementCreator;
+import uk.gov.ons.census.action.model.dto.FulfilmentMapperDTO;
+import uk.gov.ons.census.action.utility.FulfilmentMapper;
+
+public class FulfilmentsUpdatedTest {
+  private final JdbcTemplate jdbcTemplate = mock(JdbcTemplate.class);
+
+  @Test
+  public void TestAddBatchIdAndQuantity() {
+    FulfilmentMapperDTO fulfilmentMapperDTO = new FulfilmentMapperDTO();
+    fulfilmentMapperDTO.setCount(1);
+    fulfilmentMapperDTO.setFulfilmentCode("P_OR_H1");
+    List<FulfilmentMapperDTO> fulfilmentsToSend = new ArrayList<>();
+    fulfilmentsToSend.add(fulfilmentMapperDTO);
+
+    when(jdbcTemplate.query(any(String.class), any(FulfilmentMapper.class)))
+        .thenReturn(fulfilmentsToSend);
+
+    // When
+    FulfilmentProcessor fulfilmentProcessor = new FulfilmentProcessor(jdbcTemplate);
+    fulfilmentProcessor.addFulfilmentBatchIdAndQuantity();
+
+    // Then
+    verify(jdbcTemplate)
+        .update(
+            any(String.class),
+            eq(fulfilmentMapperDTO.getCount()),
+            any(UUID.class),
+            eq(fulfilmentMapperDTO.getFulfilmentCode()));
+  }
+
+  @Test
+  public void TestNoFulfilmentsInDatabase() {
+
```

Comment:
I'd prefer the test to be explicit in returning an empty list of fulfilments (and maybe named whenNoFulfilmentsAreInDatabase).

Also again test that it calls the correct methods 'correctly' and that it 

```suggestion
  @Test
  public void TestNoFulfilmentsInDatabase() {
    List<FulfilmentMapperDTO> emptyFulfilmentsList = new ArrayList<>();
    when(jdbcTemplate.query(any(String.class), any(FulfilmentMapper.class)))
        .thenReturn(emptyFulfilmentsList);

    // When
    FulfilmentProcessor fulfilmentProcessor = new FulfilmentProcessor(jdbcTemplate);
    fulfilmentProcessor.addFulfilmentBatchIdAndQuantity();

    // Then
    verify(jdbcTemplate).query(eq(EXPECTED_QUERY), any(FulfilmentMapper.class));

    verifyNoMoreInteractions(jdbcTemplate);
  }
```

+ see suggested change for making FulfilmentMapper a @component